eclipsePlugin/build.gradle

```diff
@@ -34,6 +36,12 @@ def eclipsePluginId = 'com.github.spotbugs.plugin.eclipse'
 def localProps = new Properties()
 localProps.load(new FileInputStream("$projectDir/local.properties"))
 
+ext {
+  eclipseRootDir = localProps.getProperty('eclipseRoot.dir')
+  eclipseExecutable = eclipseRootDir + (Os.isFamily(Os.FAMILY_MAC) ? "/Contents/MacOS/eclipse" : "/eclipse")
+  eclipsePluginsDir = eclipseRootDir + (Os.isFamily(Os.FAMILY_MAC) ? "/Contents/Eclipse/plugins" : "/plugins")
+}
```

Comment:
> I guess it's better to check not OS but existence of files like below:

Makes sense. While I don’t think it will make the Gradle script more portable (the `MacOS` in `/Contents/MacOS/eclipse` is pretty telling), it would support building against a pre-Mars Eclipse. Also, you could at least cross-compile against an Eclipse installation for a different OS (`./gradlew :eclipsePlugin:check`), even if you cannot build the update site (part of `./gradlew :eclipsePlugin:build`), as that would require executing the `${eclipseExecutable}`.

I will modify my pull request accordingly.

> Then why we do not drop support for Kepler, Luna and Mars?

Because a lot of them are still used in the wild. Here are some numbers from the Eclipse Code Recommenders to back this up:

In the last 30 days, there have these many downloads of the recommendation model for the JRE itself (the one library every Java project has to use) using the different Eclipse versions:

Eclipse Version | Downloads
-|-
Neon | 493,589
Mars | 222,210
Luna | 48,596
Kepler | 14,557

As the model is downloaded only once per workspace, these numbers are **lower bounds** on the number of **new workspaces** created using the various Eclipse versions; there may in fact be many more users who happily use Kepler, but haven’t felt the need to create a new workspace in months (and thus haven’t downloaded new models).

I hope you find these numbers useful.